packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -192,7 +192,9 @@ export function set_attributes(element, prev, next, lowercase_attributes, css_ha
 						events.push([
 							key,
 							value,
-							() => (next[key] = create_event(event_name, element, value, opts))
+							// since key is a var, when calling this in a later point, it will point the last value it had
+							// hence we can't do that and we have to do this instead
+							() => (next[`on${event_name}`] = create_event(event_name, element, value, opts))
```

Comment:
This will be wrong when using capture events. I suggest making `key` a `let` variable instead.
Also this needs a changeset.

Reply:
I'm sorry I don't see where it could cause a problem. The only place `key` is referenced inside a callback was this line. I also did consider about making it  a `const` (it never changes within that scope) but I thought maybe using the `var key` decalared above was intentional because of performance reasons - like not creating a scoped varibale on each iteration.

If its not the case do you agree with changing the for to:
```js
for (const key in next) {
```

Reply:
Whoops i see it now 😂😂

Reply:
well now that I actually have eyeballs that see that if-branch :D since `event_name` is basically `key` with `on` substracted from it:
I think [These lines](https://github.com/FoHoOV/svelte/blob/41ce59f38ef93581897b68a9a86fb3b328daca85/packages/svelte/src/internal/client/dom/elements/attributes.js#L175-L179) should also be changed to
```javascript
if (
event_name.endsWith('capture') &&
event_name !== 'gotpointercapture' &&
event_name !== 'lostpointercapture'
) {
event_name = event_name.slice(0, -7);
opts.capture = true;
}
```.
Let me test that real quick.

Reply:
yes the second issue highlited here is also true, will create a separete PR for that. sorry it took me some time I just got home now :D

Reply:
actually "gotpointercapture" and "lostpointercapture" don't work at all, I've created an issue (#11789) for it and the PR for it.